### PR TITLE
[CIS-1294] Module Stability & XCFrameworks

### PR DESCRIPTION
**Pull request by polqf**
### 🔗 Issue Link
https://stream-io.atlassian.net/browse/CIS-1294

#### Intermediate PRs:
[CIS-1323] Embed third party libraries #1618
[CIS-1324] Fastlane lane to create xcframeworks #1619
[CIS-1322] Embed 3rd party dependencies without framework target wrapper #1624
[CIS-1301] Automate update of 3rd party dependencies #1625
[CIS-1304, CIS-1305, CIS-1306, CIS-1307] Verify XCFrameworks integration #1637
[CIS-1354] Pipeline to distribute Static Libraries #1643
[CIS-1355, CIS-1356, CIS-1366] Use OSS as default integration source #1658

### 🎯 Goal

https://github.com/GetStream/stream-chat-swift/discussions/1614
The end goal of this initiative is to distribute our libraries as pre-built XCFrameworks, while also allowing anyone to build them themselves if needed. This enables us to reach Module Stability

### 🛠 Implementation

Detailed here: https://www.notion.so/stream-chat-fe-sdk/Module-stability-7aaa6f4f641647f1850008acc17a7227

### 🧪 Testing

The integration has been tested using all the possible integration options (Including the Dependency Managers we support)

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

[CIS-1323]: https://stream-io.atlassian.net/browse/CIS-1323?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CIS-1324]: https://stream-io.atlassian.net/browse/CIS-1324?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CIS-1322]: https://stream-io.atlassian.net/browse/CIS-1322?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CIS-1301]: https://stream-io.atlassian.net/browse/CIS-1301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CIS-1354]: https://stream-io.atlassian.net/browse/CIS-1354?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: StreamChat.xcodeproj/project.pbxproj
Comment (bielikb) [inline StreamChat.xcodeproj/project.pbxproj:10315]: Will this change keep the `Nuke` reference in the project?

File: StreamChatArtifacts.json
Comment (bielikb) [inline StreamChatArtifacts.json:2]: Let's get rid of this "test" release here.

File: StreamChatUI-XCFrameworks.podspec
Comment (bielikb) [inline StreamChatUI-XCFrameworks.podspec:12]: xcframeworks will be built by Xcode 13.0 command line tools, let's reflect the correct Swift version here

File: Tests/Shared/CustomAssertions/AssertAsync.swift
Comment (bielikb) [inline Tests/Shared/CustomAssertions/AssertAsync.swift:339]: This will probably cause issues for projects building on Xcode versions below 13.0

File: StreamChat.xcodeproj/project.pbxproj
Comment (polqf) [reply StreamChat.xcodeproj/project.pbxproj:10315]: Nuke is used in the DemoApp, as well as other example projects. The idea behind this is to find any possible clashes

File: StreamChatArtifacts.json
Comment (polqf) [reply StreamChatArtifacts.json:2]: Good catch

File: StreamChatUI-XCFrameworks.podspec
Comment (polqf) [reply StreamChatUI-XCFrameworks.podspec:12]: True that!

File: Tests/Shared/CustomAssertions/AssertAsync.swift
Comment (polqf) [reply Tests/Shared/CustomAssertions/AssertAsync.swift:339]: Hmmm, that could be true. Reverting

File: StreamChat.xcodeproj/project.pbxproj
Comment (bielikb) [reply StreamChat.xcodeproj/project.pbxproj:10315]: Ah. ok. I see your point ;)

**Comment by bielikb** [comment]
> ### 🚨 This change increases app size by 130.0 kB
> [Image of diff](https://docs.emergetools.com/docs/pr-comment)
> 
>  🗂 See the Emerge breakdown
> 🔎 [See the full analysis (4d73fdf)](https://www.emergetools.com/app/io.getstream.iOS.ChatDemoApp/type/release/build/ea961571-141c-4dbc-bc59-505bc0ef2fdf?utm_source=comment&utm_medium=link) merging into [main (5d2b950)](https://www.emergetools.com/app/io.getstream.iOS.ChatDemoApp/type/release/build/16fe192c-1255-4e25-8e3c-deb468d70d38?utm_source=comment&utm_medium=link)
> 
> > Provide a base_sha with your upload to ensure diffs are correct. [Read more in the docs](https://docs.emergetools.com/docs/uploading-basics#branch-builds)


@polqf  Let's blacklist the folders containing 3rd party code from `codecov.yml` for now. My hope here's it'll bring the 85% code cov back.

**Comment by polqf** [comment]
@bielikb it was already done 🤷 https://github.com/GetStream/stream-chat-swift/pull/1665/commits/0814dde53d8f5044afec9e84a21a47cab57c0c2b I am not sure why it is still complaining

**Comment by polqf** [comment]
It seems like Codecov only reports 0.5% relative changes, but it is using the absolute changes (85%) to block the PR 🤔

**Review by bielikb — APPROVED** [review]
Let the `library evolution` begin!

**Review by martinmitrevski — CHANGES_REQUESTED** [review]
It seems the SwiftUI SDK doesn't build after these changes. Let's resolve this before merging.

**Review by martinmitrevski — APPROVED** [review]
The issues in SwiftUI were happening locally, because of some caching state. Can be merged to main.

**Pull request merged**
